### PR TITLE
Name field var_C8 of vehicle struct. Refactor to use it as uint32

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/openrct2/GameStateSnapshots.cpp [hunk 1]
```diff
@@ -364,8 +364,7 @@ struct GameStateSnapshots : public IGameStateSnapshots
         {
             COMPARE_FIELD(rct_vehicle, pad_C6[i]);
         }
-        COMPARE_FIELD(rct_vehicle, var_C8);
-        COMPARE_FIELD(rct_vehicle, var_CA);
+        COMPARE_FIELD(rct_vehicle, additional_animaition_var);
```
Comment (AaronVanGeffen) [inline src/openrct2/GameStateSnapshots.cpp]:
```suggestion
        COMPARE_FIELD(rct_vehicle, additional_animation_var);
```

File: src/openrct2/rct1/RCT1.h [hunk 2]
```diff
@@ -306,8 +306,15 @@ struct rct1_vehicle : RCT12SpriteBase
...
     uint8_t pad_C6[0x2];
-    uint16_t var_C8;
-    uint16_t var_CA;
+    union
+    {
+        uint32_t additional_animation_var;
+        struct
+        {
+            uint16_t var_C8;
+            uint16_t var_CA;
```
Comment (ZehMatt) [inline src/openrct2/rct1/RCT1.h]: Do we still need this variable? It looks like it was only ever used a temporary variable.

File: src/openrct2/rct1/RCT1.h [hunk 3]
```diff
@@ -306,8 +306,15 @@ struct rct1_vehicle : RCT12SpriteBase
...
     uint8_t animation_frame;
     uint8_t pad_C6[0x2];
-    uint16_t var_C8;
-    uint16_t var_CA;
+    union
+    {
+        uint32_t additional_animation_var;
+        struct
+        {
+            uint16_t var_C8;
```
Comment (Gymnasiast) [inline src/openrct2/rct1/RCT1.h]: Is it necessary to split this out? If nothing directly references it any more, a comment would do IMO.

File: src/openrct2/ride/Vehicle.cpp [hunk 4]
```diff
@@ -7354,17 +7353,16 @@ static void vehicle_update_additional_animation(rct_vehicle* vehicle)
...
-                if (vehicle->var_C8 <= 0xCCCC)
+                if (vehicle->additional_animation_var <= 0xCCCC)
                 {
-                    vehicle->var_C8 += 0x3333;
+                    vehicle->additional_animation_var += 0x3333;
                 }
                 else
                 {
-                    vehicle->var_C8 += 0x3333;
+                    vehicle->additional_animation_var = 0;
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp]: Why this change?

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp]: Previously this relied on rolling over. This just makes it explicit that it is resetting to 0.

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp]: What if the previously value is not exactly 0xCCCC, but higher? It wouldn't be 0 then.

Comment (spacek531) [reply src/openrct2/ride/Vehicle.cpp:7362]: Current behavior:

On the first cycle, it updates the frame when animationState is 0xFFFF (0x3333 + 0xCCCC). 0x3333 + 0xFFFF is 0x3332.
On the next cycle, it updates the frame when animationState is 0xFFFE (0x3333 + 0xCCCB), and decreases by one every future update.

After 0x3333 frame updates, animationState will finally reach zero. I was concerned that it might insert a 6th frame somewhere, but I think it won't do that.

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp:7362]: Even if it did insert that extra frame every 0x3333 cycles it would be silly to keep it like that.

**Comment by duncanspumpkin** [comment]
Superseeded by #15267
